frontend/src/components/world/MazeGame.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactNode } from 'react';
+import { Box, useToast, Button, RenderProps, SimpleGrid } from '@chakra-ui/react';
+
+export default function MazeGame(): JSX.Element {
+  const INVITING_PLAYER = 'player1'; // TODO: This is just a placeholder, remove when invite logic gets added
+  const toast = useToast();
+
+  function renderToast(renderProp: RenderProps): ReactNode {
+    function acceptRace() {
+      renderProp.onClose();
+      // TODO: add logic to start game
+    }
+  
+    function rejectRace() {
+      renderProp.onClose();
+      toast({
+        title: "Race rejected",
+        description: "Maybe next time :(",
+        status: "info",
+        duration: 2000,
+        isClosable: true
+      });
+      // TODO: add logic to inform other player that the race has been rejected
+    }
+
+    return (<Box color="white" p={3} bg="cyan.500">
+              {INVITING_PLAYER} has invited you to race, accept?
+              <div>
+                <SimpleGrid columns={2} spacing={10}>
+                  <Button colorScheme="blackAlpha" onClick={acceptRace}>accept</Button>
+                  <Button colorScheme="whiteAlpha" onClick={rejectRace}>reject</Button>
+                </SimpleGrid>
+              </div>
+            </Box>);
+  }
```

Comment:
Could consider using a `Modal` instead of a `Box` for a slightly more inherently styled menu that shows up in the middle of the screen. https://chakra-ui.com/docs/overlay/modal

Reply:
Wouldn't a modal be a little intrusive?  If you're talking to someone and someone sends you an invite, the whole screen gets filled up by the popup.  Lmk if that's not how modals work though I'm just judging from examples.

Reply:
Hmmmmmm yes that is intrusive, but isn't the point of an invite to be intrusive? The majority of the screen is taken up by the map, and if you're talking to people I would assume you're not actively moving about anyhow, so it's fine for the map to get obscured while the invite draws your attention.

Reply:
But other options could be making a custom toast with buttons inside or figuring out complex styling to make the modal smaller/less intrusive, or just leaving the `Box` as is lol, this suggestion is optional

Reply:
Wait it is a custom toast with buttons right now, unless there's another way to make custom toasts that I'm not aware of?

Reply:
I see your point though, I can play around with modals later tonight to see how it feels and add an update if its seems ok.

Reply:
Wait @lindojong can you choose which components become inert when the modal pops up or does modal automatically make everything behind it inert and grayed out?